src/renderers/dom/shared/__tests__/DOMPropertyOperations-test.js

```diff
@@ -83,181 +64,105 @@ describe('DOMPropertyOperations', () => {
           return '<html>';
         },
       };
-      DOMPropertyOperations.setValueForProperty(stubNode, 'role', obj);
-      expect(stubNode.getAttribute('role')).toBe('<html>');
+      var container = document.createElement('div');
+      ReactDOM.render(<div role={obj} />, container);
+      expect(container.firstChild.getAttribute('role')).toBe('<html>');
     });
 
     it('should not remove empty attributes for special properties', () => {
-      stubNode = document.createElement('input');
-      ReactDOMComponentTree.precacheNode(stubInstance, stubNode);
-
-      DOMPropertyOperations.setValueForProperty(stubNode, 'value', '');
+      var container = document.createElement('div');
+      ReactDOM.render(<input value="" />, container);
       // JSDOM does not behave correctly for attributes/properties
-      //expect(stubNode.getAttribute('value')).toBe('');
-      expect(stubNode.value).toBe('');
+      //expect(container.firstChild.getAttribute('value')).toBe('');
+      expect(container.firstChild.value).toBe('');
     });
 
     it('should remove for falsey boolean properties', () => {
-      DOMPropertyOperations.setValueForProperty(
-        stubNode,
-        'allowFullScreen',
-        false,
-      );
-      expect(stubNode.hasAttribute('allowFullScreen')).toBe(false);
+      var container = document.createElement('div');
+      ReactDOM.render(<div allowFullScreen={false} />, container);
+      expect(container.firstChild.hasAttribute('allowFullScreen')).toBe(false);
     });
 
     it('should remove when setting custom attr to null', () => {
-      DOMPropertyOperations.setValueForProperty(stubNode, 'data-foo', 'bar');
-      expect(stubNode.hasAttribute('data-foo')).toBe(true);
-      DOMPropertyOperations.setValueForProperty(stubNode, 'data-foo', null);
-      expect(stubNode.hasAttribute('data-foo')).toBe(false);
-    });
-
-    it('should use mutation method where applicable', () => {
```

Comment:
👍. Long term, I believe we can remove mutation method all together if we circle back to https://github.com/facebook/react/pull/10150